src/test/java/test/listeners/AlterSuiteListenerTest.java

```diff
@@ -17,35 +17,65 @@
 
     @Test
     public void executionListenerWithXml() {
-        XmlSuite suite = runTest(AlterSuiteListener1SampleTest.class, AlterSuiteNameListener.class.getName());
+        XmlSuite suite = runTest(AlterSuiteListener1SampleTest.class, AlterSuiteNameListener.class.getName()).second();
         Assert.assertEquals(suite.getName(), AlterSuiteNameListener.class.getSimpleName());
     }
 
     @Test
     public void executionListenerWithoutListener() {
-        XmlSuite suite = runTest(AlterSuiteListener1SampleTest.class, null/*Donot add the listener*/);
+        XmlSuite suite = runTest(AlterSuiteListener1SampleTest.class).second();
         Assert.assertEquals(suite.getName(), ALTER_SUITE_LISTENER);
     }
 
     @Test
     public void executionListenerWithXml2() {
-        XmlSuite suite = runTest(AlterSuiteListener1SampleTest.class, AlterXmlTestsInSuiteListener.class.getName());
+        XmlSuite suite = runTest(AlterSuiteListener1SampleTest.class, AlterXmlTestsInSuiteListener.class.getName()).second();
         Assert.assertEquals(suite.getTests().size(), 2);
     }
 
+    @Test(description = "GITHUB-2469")
+    public void executionListenerWithXml3() {
+        Pair<TestNG, XmlSuite> retObjects = runTest(AlterSuiteListener1SampleTest.class,
+                AlterXmlTestWithParameterInSuiteListener.class.getName(), AlteredXmlSuiteReadListener.class.getName());
+        TestNG tng = retObjects.first();
+        XmlSuite suite = retObjects.second();
+        Assert.assertEquals(suite.getTests().size(), 2);
+        List<ISuiteListener> listeners = tng.getSuiteListeners();
+        Assert.assertNotNull(listeners);
+        if(listeners.size() > 0){
```

Comment:
@aditya-qapitol - I think we should refrain from using these conditional assertions. 

We should instead refactor this such that 

```
List<ISuiteListener> listeners = Optional.ofNullable(tng.getSuiteListeners()),orElse(new ArrayList<>());
```
That way we dont need to use `Assert.assertNotNull()` and we should just use `Assert.assertFalse(listeners.isEmpty())` and thus get rid of `if (listeners.size() > 0)` conditional statement.